iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -144,6 +145,12 @@ public static Table createTable(Configuration conf, Properties props) {
 
     Map<String, String> map = filterIcebergTableProperties(props);
 
+    // Hive only supports merge-on-read delete mode, it will actually throw an error if DML operations are attempted on
+    // tables that don't have this (the default is copy-on-write)
+    if ("2".equals(props.get(TableProperties.FORMAT_VERSION))) {
+      map.put(TableProperties.DELETE_MODE, "merge-on-read");
```

Comment:
Will this be deleted when copy-on-write is supported later? If so, it will be good to add a minor TODO here.